CONTRIBUTING.md

```diff
@@ -29,3 +29,99 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+and it is expected the changes from step 2 will also be merged soon after. Before merging into Parquet-java a parquet-format release
+must be performed.
+
+#### General guidelines/preferences on additions.
+
+1. To the greatest extent possible changes should have an option for forwards compatibility (old readers can still read files).
+2. New encodings should be fully specified in this repository and ideally not rely on an external dependencies for implementation (i.e. Parquet is the source of truth for the encoding).
+3. New compression mechanisms must have a pure Java implementation that can be used as dependency in parquet-java.
+
+### Releases
+
+The Parquet community aims to do releases of the format package only as needed when new features are introduced.
+If multiple new features are being proposed simultaneously some features might be consolidated into the same release.  Guidance is provided below on when implementations should enable features added to the specification.
+Due to confusion in the past over parquet versioning it is not expected that there will be a 3.0 release of the specification in the foreseeable future.
+
+### Compatibility and Feature Enablement
+
+For the purposes of this discussion we classify features into the following buckets:
+
+1. Backwards compatible.  A file written by an older version of a library can be read by a newer version of the
+library.
+
+2. Forwards compatible.  A file written by a new version of the library can be read by an older version
+of the library. 
+
+3. Forward compatible with suboptimal performance. A file written by a new version of the library can
+be read an older version of the library but performance might be suboptimal (e.g. statistics might be missing
+from the older reader's perspective).
+
+4. Forward incompatible. A file written with a new version of the library cannot be read by an older version
+of the library.
+
+The Parquet community hopes that new features are widely beneficial
+to users of Parquet, and therefore third-party implementations will
+adopt them quickly after they are introduced. It is assumed that most new features will be implemented behind a feature flag that defaults to "off".To avoid, compatibility issues across the ecosystem some amount of lead time is desirable to ensure a critical mass of Parquet implementations support a feature.  Therefore, the Parquet PMC gives the following guidance for changing a feature to be "on" by default:
+
+1. Backwards compatibility is the concern of implementations but given the ubiquity of Parquet and the length
+of time it has been used, libraries SHOULD support reading older
+file variants.
+
+2. Forward compatible changes MAY be used by default in implementations once the parquet-format containing
+those changes has been formally released. These features SHOULD be turned on 1 year after the parquet-java
+implementation containing feature is released (e.g. it is expected
```

Comment:
I don't think this "SHOULD" is really warranted. While the Java implementation is still the "primary" one from a community POV, that doesn't mean that everything that the Java implementation supports should be blindly turned on by default.

Reply:
@pitrou I changed a bunch of wording around to be less prescriptive, please let me know if this addresses your concerns.

Reply:
Per this feedback and @alamb I removed all upper case SHOULD/MUST/MAY statements to avoid confusion.